sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/finishAnalysis.scala

```diff
@@ -66,6 +66,34 @@ object RewriteNonCorrelatedExists extends Rule[LogicalPlan] {
   }
 }
 
+/**
+ * Wrap complex grouping expression in aggregate expressions without aggregate function into
```

Comment:
What does "complex" means in this context?

Reply:
Described in https://github.com/apache/spark/pull/31913/commits/f46b89d6d81be4a7867fd3bc9edeccc1f7e9aea5